build.gradle

```diff
@@ -48,11 +48,20 @@ subprojects {
     strictCheck true
   }
 
+  configurations {
+    all {
+      exclude group: 'org.slf4j', module: 'slf4j-log4j12'
+    }
+  }
+
   plugins.withType(JavaPlugin) {
     project.apply(plugin: 'checkstyle')
 
     dependencies {
+      compile 'org.slf4j:slf4j-api:1.7.25'
+
       testCompile 'org.testng:testng:6.11'
+      testCompile 'org.slf4j:slf4j-simple:1.7.25'
```

Comment:
Is this required by all modules? I think it is required for the plugin. Depending on the answer of this, should we relocate the above section as well?

Reply:
I think we should include the `slf4j-simple` binding by default in our test compile path for all modules. This allows any dependencies which use `slf4j-api` to correctly print out logging statements during test cases even though Transport explicitly maybe not be using it.
Yes `slf4j-api` is probably not needed by all modules, but I thought I would just keep it here. ideally we would have logging statements in all our modules and most libraries have `slf4j-api` exported in their dependency graph anyway so it being there is not a surprise. Let me know if you would like to change that.

Reply:
Removed `slf4j-api` turns out we don't need it. We have no log lines. The only one in TransportPlugin uses Gradle's logging libraries.